src/intern.rs

```diff
@@ -4,6 +4,9 @@ use std::sync::RwLock;
 use lasso::{Rodeo, Spur};
 use lazy_static::lazy_static;
 
+/// A opaque identifier for a string which has been [interned].
```

Comment:
Maybe go into a little more detail on _why_ strings are being interned, someone not familiar may not understand the reasoning at a glance